### PR TITLE
Willdle dev

### PR DESCRIPTION
**Pull request by willle3456**
Adding ros_control integration. Also, grSim communication works

### CODE REVIEW

**Comment by willle3456** [comment]
This should resolve #1

File: src/robocup_control/src/drive_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+namespace robocup_control
+{
+bool DriveController::init(hardware_interface::VelocityJointInterface* hw, ros::NodeHandle& n)
+{
+  ROS_INFO("Initializing drive controller");
+  double pub_rate = 0.0f;
+  
+  // effectively a constant
+  double pi = 3.14159265;
+  n.getParam("/robocup_control/drive_controller/joints", joint_names);
```
Comment (Gomer1800) [inline src/robocup_control/src/drive_controller.cpp:22]: explain purpose of line

File: src/robocup_control/src/drive_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  
+  // get publish rate here
+  n.getParam("/robocup_control/joint_state_controller/publish_rate", pub_rate);
+  update_rate = 1 / pub_rate;
+
+  // save the publish rate in a variable to get the right update time
+  num_joints = joint_names.size();
+  drive_motors.resize(num_joints);
+
```
Comment (Gomer1800) [inline src/robocup_control/src/drive_controller.cpp:31]: Explain why push_back is being called

File: src/robocup_control/include/robocup_control/drive_controller.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  double angle;
+
+  double update_rate;
+  std::vector<hardware_interface::JointHandle> drive_motors;
+  std::vector<double> motor_angles;
+  std::vector<std::string> joint_names;
+  int num_joints;
+  ros::Subscriber speed_sub;
+  
+  // subscriber call back for new commanded x, y, theta velocities
```
Comment (Gomer1800) [inline src/robocup_control/include/robocup_control/drive_controller.h:54]: no Doxygen comment?

**Review by Gomer1800 — CHANGES_REQUESTED** [review]

**Pull request merged**